awx/main/models/unified_jobs.py

```diff
@@ -911,6 +915,29 @@ def result_stdout_text(self, value):
         related.result_stdout_text = value
         related.save()
 
+    @property
+    def event_parent_key(self):
+        tablename = self._meta.db_table
+        return {
+            'main_job': 'job_id',
+            'main_adhoccommand': 'ad_hoc_command_id',
+            'main_projectupdate': 'project_update_id',
+            'main_inventoryupdate': 'inventory_update_id',
+            'main_systemjob': 'system_job_id',
+        }[tablename]
+
+    def get_event_queryset(self):
+        return self.event_class.objects.filter(**{self.event_parent_key: self.id})
+
+    @property
+    def events_processed(self):
```

Comment:
Could we rename this to make it clear it's a boolean?  My brain read this property name expecting back a count.  Maybe something like `def event_processing_finished(self):` ?

Reply:
The name here has to be the same as what we're returning in the API.

That said, yes, I'm in favor of a better name for the field. That's a little long though...

 - events_have_processed
 - finished_processing
 - finalized
 - can_haz_events

Reply:
I like `events_have_processed`

Reply:
I might stereotype `events_have_processed` as an integer. I think `event_processing_finished` is a little more bool-ish, and is my current favorite.